pkg/tools/closectx/context.go

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2021 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package closectx provides helper method to create Close/Unregister context in case of Request/Register failure
+package closectx
```

Comment:
Could we name this something less bound to Close and more reflecting the fact that we are creating a mechanism *generally* appropriate to allow us to reasonably do things like Close after ourselves, and make other sorts of calls that start at some point in the future...

Reply:
Probably we can name it something like `postponectx`? And so description would be:
```
// Package postponectx provides helper method to postpone context deadline.
```

Reply:
This is moving in the right direction :)

As a side note: I don't like the current name *I'm* using (ctxFunc) all that much either... so please don't read to much into it :)

Reply:
Perhaps in the vein of `postponectx` we could have package `postpone` and the call be something like `postpone.Context(ctx)`

Reply:
Done.